src/logger.cpp

```diff
@@ -10,21 +10,19 @@
 namespace diskann
 {
 
+#ifdef ENABLE_CUSTOM_LOGGER
 DISKANN_DLLEXPORT ANNStreamBuf coutBuff(stdout);
 DISKANN_DLLEXPORT ANNStreamBuf cerrBuff(stderr);
 
 DISKANN_DLLEXPORT std::basic_ostream<char> cout(&coutBuff);
 DISKANN_DLLEXPORT std::basic_ostream<char> cerr(&cerrBuff);
-
-#ifdef ENABLE_CUSTOM_LOGGER
```

Comment:
The intention of ENABLE_CUSTOM_LOGGER is to allow prod env set a custom logger in `g_logger.`

The default behavior (i.e., without ENABLE_CUSTOM_LOGGER) is to use stdout and stderror logging which is what the line 14-line18 does?

So these global variables seems to be an issue?